WordPress/Classes/ViewRelated/Blog/My Site/MySiteViewController+OnboardingPrompt.swift

```diff
@@ -0,0 +1,48 @@
+import Foundation
+
+extension MySiteViewController {
+    func startObservingOnboardingPrompt() {
+        NotificationCenter.default.addObserver(self, selector: #selector(onboardingPromptWasDismissed(_:)), name: .onboardingPromptWasDismissed, object: nil)
+    }
+
+    @objc func onboardingPromptWasDismissed(_ notification: NSNotification) {
+        guard
+            let userInfo = notification.userInfo,
+            let option = userInfo["option"] as? OnboardingOption
+        else {
+            return
+        }
+
+        switch option {
+        case .stats:
+            // Show the stats view for the current blog
+            // We have to switch to the "Site Menu" first
+            switchTab(to: .siteMenu)
```

Comment:
Yeah, sadly I hit some issues when I tried to show the stats:

- When I tried to use `StatsViewController.show` the stats view is pushed but it just shows everything as an error. 
- If I try to use the method above without switching the tab first nothing happens. 

Not sure why that is, but this seems to work okay for now.

Reply:
@emilylaguna I'm a little bit concerned about that because it will affect the dashboard experiment. Also, this is exactly the flow to which we assign the experiment to users.

Even if we take into account that the experiment might end by the time this is released, we might want to enable the dashboard as the default for all users, and this will flip the first screen they see if they select Stats.

I tried changing the code to this:

```swift
case .stats:
  StatsViewController.show(for: blog!, from: self)
```

Tested for both site menu and the dashboard as the default and it worked just fine in both cases. :)

Reply:
I tried that but I am experiencing inconsistent results with it. It seems half the time it loads with an error:

<img src="https://user-images.githubusercontent.com/793774/165346353-4a95d47c-763e-4057-8202-5f0c4c8a7c0b.png" width="160" />


I'm going to see if I can nail down why.

Reply:
Spent some time on this, I think this happens due to the stats insights caching and me repeatedly logging in and out and the cache not resetting itself. When I delete the app try again I don't see this happening. 

Changed the code here: 3f6e60fd4ae46c07beaa2e32b4c5371e867bfb8d